lib/Backend/FlowGraph.cpp

```diff
@@ -25,27 +147,31 @@ void
 FlowGraph::Build(void)
 {
     Func * func = this->func;
-
     BEGIN_CODEGEN_PHASE(func, Js::FGPeepsPhase);
     this->RunPeeps();
     END_CODEGEN_PHASE(func, Js::FGPeepsPhase);
 
     // We don't optimize fully with SimpleJit. But, when JIT loop body is enabled, we do support
     // bailing out from a simple jitted function to do a full jit of a loop body in the function
-    // (BailOnSimpleJitToFullJitLoopBody). For that purpose, we need the flow from try to catch.
-    if (this->func->HasTry() &&
-        (this->func->DoOptimizeTryCatch() ||
-            (this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody())
-        )
-       )
+    // (BailOnSimpleJitToFullJitLoopBody). For that purpose, we need the flow from try to handler.
+    if (this->func->HasTry() && (this->func->DoOptimizeTryCatch() ||
+            (this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody())))
     {
         this->catchLabelStack = JitAnew(this->alloc, SList<IR::LabelInstr*>, this->alloc);
     }
+    if (this->func->HasFinally() && (this->func->DoOptimizeTryFinally() ||
+        (this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody())))
```

Comment:
Oh, I was looking at the 2nd iteration - second issue seems  fixed in 3rd iteration :). We should still hoist the common conditions though

---
In reply to: [116533348](https://github.com/Microsoft/ChakraCore/pull/2954#discussion_r116533348) [](ancestors = 116533348)